File: awesome-calendar/src/main/java/com/archit/calendardaterangepicker/customviews/CalendarListener.java
package com.archit.calendardaterangepicker.customviews;

import androidx.annotation.NonNull;

import java.util.Calendar;

public interface CalendarListener {
    /**
     * Called on first date selection.
     * @param startDate First selected date.
     */
    void onFirstDateSelected(@NonNull final Calendar startDate);

    /**
     * Called on first and last date selection.
     * @param startDate First date.
     * @param endDate Last date.
     */
    void onDateRangeSelected(@NonNull final Calendar startDate, @NonNull final Calendar endDate);
}

File: awesome-calendar/src/main/java/com/archit/calendardaterangepicker/customviews/CalendarListener.kt
package com.archit.calendardaterangepicker.customviews

import java.util.Calendar

interface CalendarListener {
    /**
     * Called on first date selection.
     * @param startDate First selected date.
     */
    fun onFirstDateSelected(startDate: Calendar)

    /**
     * Called on first and last date selection.
     * @param startDate First date.
     * @param endDate Last date.
     */
    fun onDateRangeSelected(startDate: Calendar, endDate: Calendar)
}